### PR TITLE
Added top consumer to ceph storage plugin

### PR DESCRIPTION
**Pull request by cloudbehl**
Screenshot from 2019-07-16 18-52-46

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.ceph-top-consumer-card__dropdown {
+  display: flex;
+  margin: 0.6em 0em;
+}
+.ceph-top-consumer-card__dropdown-item {
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.scss]: add empty line between classes

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ChartLineIcon } from '@patternfly/react-icons';
+import {
+  ChartGroup,
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx]: alpha (+ all other imports)

File: frontend/packages/ceph-storage-plugin/src/utils/index.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+export const getMetricType = (resource, metricType) => _.get(resource, ['metric', metricType], '');
+
+export const formatToShortTime = (timestamp: Date): string => {
+  const dt = new Date(timestamp);
+  // returns in HH:MM format
+  return dt.toString().substring(16, 21);
+};
+
+export const getGraphVectorStats: GetStats = (response, metricType, unit) => {
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/utils/index.ts]: you should use `getInstantVectorStats` from `graph/utils`. You also dont need to specify `GetStats` as it is already in `graph/utils` too

File: frontend/packages/ceph-storage-plugin/src/utils/index.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import * as _ from 'lodash';
+import { humanizeBinaryBytesWithoutB } from '@console/internal/components/utils';
+import { PrometheusResponse, DataPoint } from '@console/internal/components/graphs';
+
+export const getMetricType = (resource, metricType) => _.get(resource, ['metric', metricType], '');
+
+export const formatToShortTime = (timestamp: Date): string => {
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/utils/index.ts]: you should use `twentyFourHourTime` from `utils/datetime` to be consistent with rest of the console

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    StorageDashboardQuery[TopConsumerResourceValue[metricType] + TopConsumerSortByValue[sortBy]]
+    ],
+    'result',
+  ]);
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Top Consumers</DashboardCardTitle>
+        <div className="ceph-top-consumer-card__dropdown">
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx]: height of the card header is inconsistent with others. Take a look at these styles https://github.com/openshift/console/pull/1722/files#diff-163991d4c119db922a729100ce2bea7fR91

File: frontend/packages/ceph-storage-plugin/src/utils/index.ts (hunk 3 above)
Comment (cloudbehl) [reply frontend/packages/ceph-storage-plugin/src/utils/index.ts]: @rawagner can't use the getInstantVectorStats as it is for a single value doesn't work when multiple values are being returned.

Also the same for GetStats as in my case multiple PrometheusResponse are being returned and that is returning only one.

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx (hunk 5 above)
Comment (rawagner) [reply frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx]: sorry, those dropdowns are actually not in header as yours are.
@andybraren @matthewcarleton any help on how to include these dropdowns into PF4 Card Header and keep header height consistent ? 

dropdowns-header

See the difference between Capacity and Top Consumers header

Comment (matthewcarleton) [reply frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx]: Hey @rawagner I believe you can use the actions area for the card header provided. That should fix the spacing isssues. If not we may need to investigate further if there is unnecessary spacing added to other elements in there.

**Pull request merged**